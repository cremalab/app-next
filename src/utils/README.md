# `./src/utils`

This directory is intended to organize global application utilitiesâ€”functions, constants, and more.

## FAQ

1. _**Are `utils` different from `helpers`, `lib`, etc.?**_ No. It's jut the name that was chosen.
2. _**What constitutes a `util`?**_ Anything that isn't a React component, type, and doesn't belong to another project-specific grouping of code (e.g. Redux related code, etc.). Application-specific logic and type transformation code are some examples of what would make a good `util`.
3. _**Must `utils` be functions?**_ No. They can be any value you need them to be, but functions tend to be the most common kind of util.
