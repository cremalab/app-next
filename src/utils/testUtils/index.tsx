import { RenderOptions, render } from "@testing-library/react"
import { ReactElement, ReactNode } from "react"

const Providers = ({ children }: { children?: ReactNode }) => {
  return <>{children}</>
}

const customRender = (ui: ReactElement, options: RenderOptions = {}) =>
  render(ui, { wrapper: Providers, ...options })

// re-export everything
// eslint-disable-next-line import/export
export * from "@testing-library/react"

// override render method
// eslint-disable-next-line import/export
export { customRender as render }
