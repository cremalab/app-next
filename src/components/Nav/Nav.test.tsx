import React from "react"
import { render } from "../../utils/testUtils"
import { Nav } from "."

describe("Nav", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Nav />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it("has the expected links", () => {
    const { getByText } = render(<Nav />, {})
    expect(getByText("Welcome")).toHaveAttribute("href", "/")
    expect(getByText("SSR")).toHaveAttribute("href", "/ssr")
    expect(getByText("SSG")).toHaveAttribute("href", "/ssg")
  })
})
