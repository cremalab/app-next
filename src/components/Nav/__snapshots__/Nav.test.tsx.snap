// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="container"
  >
    <ul
      class="list"
    >
      <li>
        <a
          class="link"
          href="/"
        >
          Welcome
        </a>
      </li>
      <li>
        <a
          class="link"
          href="/ssr"
        >
          SSR
        </a>
      </li>
      <li>
        <a
          class="link"
          href="/ssg"
        >
          SSG
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
